### PR TITLE
Like feature and page ordering features

### PR DESCRIPTION
**Pull request by A5rocks**
According to sublime merge, originally `like-feature-(and-page-ranking)` was branched from `add-page-render-functionality`, so maybe we should pull those changes back in? This should lead to having to review less lines!

Replaces https://github.com/Artemis21/summer-code-jam-2020/pull/26

### CODE REVIEW

File: jolly-jellyfish/src/page_maker/templates/page_maker/base.html [hunk 1]
```diff
@@ -35,10 +35,10 @@
...
     {% if user.is_authenticated %}
         <a href="{% url 'user-detail' pk=user.id %}"> {{ user.username }}</a>
-        <a href="{% url 'logout' %}">Logout</a>
+        <a href="{% url 'logout' %}?next={{ request.path }}">Logout</a>
     {% else %}
-        <a href="{% url 'login' %}">Login</a>
-        <a href="{% url 'register' %}">Register</a>
+        <a href="{% url 'login' %}?next={{ request.path }}">Login</a>
+        <a href="{% url 'register' %}?next={{ request.path }}">Register</a>
     {% endif %}
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/base.html]: We shouldn't indent in if blocks.

File: jolly-jellyfish/src/page_maker/templates/page_maker/webpage_detail.html [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 {% block content %}
 <div><h1>{{ webpage.user_title }}</h1></div>
+<div>
+    {{ num_likes }} like{{ num_likes|pluralize }}
+    <form method="POST" action="">
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/webpage_detail.html]: Isn't this blank `action` a hardcoded url?

File: jolly-jellyfish/src/page_maker/templates/page_maker/webpage_list.html [hunk 3]
```diff
@@ -17,12 +18,35 @@ <h2>Top pages</h2>
...
     </tr>
     {% endfor %}
 </table>
+<br>
+<div class="pagination">
+    <span class="step-links">
+        {% if page_obj.has_previous %}
+            <a href="?page=1">First page</a>
+            <a href="?page={{ page_obj.previous_page_number }}">Previous page</a>
+        {% endif %}
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/webpage_list.html]: We shouldn't be indenting here

File: jolly-jellyfish/src/page_maker/templates/page_maker/webpage_list.html [hunk 4]
```diff
@@ -17,12 +18,35 @@ <h2>Top pages</h2>
...
+
+        {% if page_obj.has_next %}
+            <a href="?page={{ page_obj.next_page_number }}">Next page</a>
+            <a href="?page={{ page_obj.paginator.num_pages }}">Last page</a>
+        {% endif %}
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/webpage_list.html]: We shouldn't be indenting here

File: jolly-jellyfish/src/page_maker/templates/page_maker/user_detail.html [hunk 5]
```diff
@@ -6,25 +6,36 @@ <h2>User {{ viewed_user.username }}</h2>
...
     {% for webpage in user_pages %}
     <tr>
         <td>
             {% if "placeholder_img" not in webpage.thumbnail.url %}
-            <img src="{{ webpage.thumbnail.url }}">
+                <img src="{{ webpage.thumbnail.url }}">
             {% else %}
-            Go to page itself to generate a thumbnail for it.
+                Go to page itself to generate a thumbnail for it.
             {% endif %}
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/user_detail.html]: We shouldn't be indenting in an if block.

File: jolly-jellyfish/src/page_maker/templates/page_maker/webpage_view.html [hunk 6]
```diff
@@ -20,8 +20,11 @@ <h1><span>{{ webpage.user_title }}</span></h1>
     {% endif %}
 </p>
 <p>{{ webpage.user_text_3 }}</p>
+
 <p>
-    <a href="{% url 'webpage-detail' pagename=webpage.name %}">Comment on my page</a>
+    {{ num_likes }} like{{ num_likes|pluralize }}
+<br>
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/webpage_view.html]:
```suggestion
    {{ num_likes }} like{{ num_likes|pluralize }}<br>
```

File: jolly-jellyfish/src/page_maker/templates/registration/register.html [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {% extends "page_maker/base.html" %}
 {% block content %}
    <div>
-       <h2>Sing Up</h2>
+       <h2>Sign Up</h2>
        <form method="POST">
            {% csrf_token %}
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/registration/register.html]: This should probably be moved down an indentation level

File: jolly-jellyfish/src/page_maker/urls.py [hunk 8]
```diff
@@ -20,7 +20,7 @@
 )
 
 urlpatterns = [
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/urls.py]: These shouldn't end with `/`.

File: jolly-jellyfish/src/page_maker/views.py [hunk 9]
```diff
@@ -107,18 +116,44 @@ def get_context_data(self, **kwargs):
...
+                context['like_response'] = "You can't like your own page!"
+            else:
+                like_obj, created = Like.objects.get_or_create(author=request.user, parent_page=self.object)
+                if created:  # this is first time the user has liked the webpage
+                    context['like_response'] = "Thanks for the like!"
+                else:
+                    like_obj.delete()
+                    context['like_response'] = "You've now unliked this page."
+        else:
+            context['like_response'] = "Please login to leave a like."
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/views.py]: Double-quotes -> single-quotes.

**Review by A5rocks — COMMENTED** [review]
A couple changes... (I can't "request changes" cause I made this PR :( )

File: jolly-jellyfish/src/page_maker/templates/page_maker/webpage_detail.html (hunk 2 above)
Comment (A5rocks) [reply jolly-jellyfish/src/page_maker/templates/page_maker/webpage_detail.html]: nevermind it doesn't seem to be?

**Review by tameTNT — APPROVED** [review]
Looks good to me - but then I wrote the stuff so I would say that 😆

File: jolly-jellyfish/src/page_maker/templates/page_maker/webpage_view.html [hunk 10]
```diff
@@ -22,9 +22,8 @@ <h1><span>{{ webpage.user_title }}</span></h1>
 <p>{{ webpage.user_text_3 }}</p>
 
 <p>
-    {{ num_likes }} like{{ num_likes|pluralize }}
-<br>
-    <a href="{% url 'webpage-detail' pagename=webpage.name %}">Like & comment on my page!</a>
+  {{ num_likes }} like{{ num_likes|pluralize }}<br>
+  <a href="{% url 'webpage-detail' pagename=webpage.name %}">Comment on my page</a>
```
Comment (tameTNT) [inline jolly-jellyfish/src/page_maker/templates/page_maker/webpage_view.html]:
```suggestion
  <a href="{% url 'webpage-detail' pagename=webpage.name %}">Like and comment on my page!</a>
```

**Review by Artemis21 — APPROVED** [review]
LGTM

**Pull request merged**